test/gateway/pollingInterval.js

```diff
@@ -488,3 +489,302 @@ test('Polling schemas (cache should be cleared)', async (t) => {
+    User: {
+      __resolveReference: (user, args, context, info) => user
+    }
+  }
+
+  const userService = Fastify()
+  const gateway = Fastify()
+
+  t.tearDown(() => {
+    userService.close()
+    gateway.close()
+  })
+
+  userService.register(GQL, {
+    schema: `
+      extend type Query {
+        me: User
+      }
+
+      extend type Subscription {
+        updatedUser: User
+      }
+
+      extend type Mutation {
+        triggerUser: Boolean
+      }
+
+      type User @key(fields: "id") {
+        id: ID!
+        name: String!
+      }
+    `,
+    resolvers: resolvers,
+    federationMetadata: true,
+    subscription: true
+  })
+
+  await userService.listen(0)
+
+  const userServicePort = userService.server.address().port
+
+  gateway.register(GQL, {
+    subscription: true,
+    gateway: {
+      services: [
+        {
+          name: 'user',
+          url: `http://localhost:${userServicePort}/graphql`,
+          wsUrl: `ws://localhost:${userServicePort}/graphql`
+        }
+      ],
+      pollingInterval: 2000
+    }
+  })
+
+  await gateway.listen(0)
+
+  const ws = new WebSocket(
+    `ws://localhost:${gateway.server.address().port}/graphql`,
+    'graphql-ws'
+  )
+
+  const client = WebSocket.createWebSocketStream(ws, {
+    encoding: 'utf8',
+    objectMode: true
+  })
+
+  t.tearDown(() => {
+    client.destroy()
+  })
+
+  client.setEncoding('utf8')
+
+  client.write(
+    JSON.stringify({
+      type: 'connection_init'
+    })
+  )
+
+  client.write(
+    JSON.stringify({
+      id: 1,
+      type: 'start',
+      payload: {
+        query: `
+          subscription {
+            updatedUser {
+              id
+              name
+            }
+          }
+        `
+      }
+    })
+  )
+
+  const promise = () =>
+    new Promise((resolve) => {
+      client.on('data', (chunk) => {
+        const data = JSON.parse(chunk)
+
+        if (data.type === 'connection_ack') {
+          gateway.inject({
+            method: 'POST',
+            url: '/graphql',
+            body: {
+              query: `
+              mutation {
+                triggerUser
+              }
+            `
+            }
+          })
+        } else if (data.type === 'data' && data.id === 1) {
+          const { payload: { data: { updatedUser = {} } = {} } = {} } = data
+
+          t.deepEqual(updatedUser, {
+            id: 'u1',
+            name: 'John'
+          })
+
+          resolve()
+        }
+      })
+    })
+
+  return promise()
+    .then(async () => {
+      userService.graphql.replaceSchema(
+        buildFederationSchema(`
+          extend type Query {
+            me: User
+          }
+
+          extend type Subscription {
+            updatedUser: User
+          }
+
+          extend type Mutation {
+            triggerUser: Boolean
+          }
+
+          type User @key(fields: "id") {
+            id: ID!
+            name: String!
+            lastName: String
+          }
+        `)
+      )
```

Comment:
I would expect the websocket connection to error/disconnect after this happens, this is not tested.

Reply:
I have added a few new checking for `ws.readyState`.
Why do you think that WebSocket should be disconnected?

It will be bad for the end-user (because he needs to refresh the page, reload the application, etc.)